SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCrashLoops.java

```diff
@@ -429,20 +434,39 @@ boolean hasCooldownExpired(
     List<Long> recentFailures,
     long bucketSizeMillis,
     int numBuckets,
-    double percentThreshold
+    double percentThreshold,
+    int minBucketIndexPercent
   ) {
     long thresholdFailTimeMillis = now - (bucketSizeMillis * numBuckets);
-    Map<Long, List<Long>> bucketedFailures = recentFailures
-      .stream()
-      .filter(t -> t > thresholdFailTimeMillis)
-      .collect(Collectors.groupingBy(e -> e / bucketSizeMillis));
-    long bucketsWithFailure = bucketedFailures
-      .entrySet()
-      .stream()
-      .filter(e -> !e.getValue().isEmpty())
-      .count();
+    long[] bucketThresholds = new long[numBuckets];
```

Comment:
Why use an array rather than a `List`, OOC?

Reply:
Ah, I was originally counting something based on the index in that next loop, then changed the implementation a bit. Doesn't particularly matter in current form. Could change to a stream -> List or something like that

Reply:
updated since List is generally more readable/flexible